package.json

```diff
@@ -9,14 +9,17 @@
   "main": "dist/next-urql.min.js",
   "module": "dist/next-urql.es.min.js",
   "source": "src/index.ts",
+  "types": "index.d.ts",
   "author": "Formidable",
   "license": "MIT",
   "scripts": {
-    "check:types": "tsc",
+    "build": "rollup -c rollup.config.js",
+    "check:types": "tsc --noEmit",
+    "clean": "rimraf ./dist ./node_modules/.cache",
     "format": "prettier --write **/*.{ts,tsx,d.ts,js}",
     "lint": "eslint **/*.{ts,tsx}",
-    "test": "jest",
-    "build": "rollup -c rollup.config.js"
+    "prepublishOnly": "run-s clean test build",
```

Comment:
Do we also want to add a `README.md` (typically we place this in a `CONTRIBUTING.md` but this repo doesn't have one) a la https://github.com/FormidableLabs/spectacle-cli/blob/master/CONTRIBUTING.md#releasing-a-new-version-to-npm

Also, perhaps consider an `npm version` workflow in lieu of `prepublishOnly`, but no biggie either way....

Reply:
Good shout, I'll add `CONTRIBUTING.md` while I'm here. I think I opted for `prepublishOnly` since it appears to be what `urql` uses. Reading through `spectacle-cli`, it looks like you're using a `preversion` script? Is there any meaningful differences between the two?

Reply:
I like `npm version` workflows better because (1) it guarantees a good version + git commit and (2) to a lesser extent the awful history of the `prepublish` script in `npm` (which was fixed in modern npm with the `prepublishOnly` lifecycle).

Also, for things like [`publishr`](https://github.com/FormidableLabs/publishr#readme) you need the extra `*version` lifecycle to get away with the whole scheme, but that's not in use here.

So, all in all, whatever works for you, that ideally guarantees:

- Good semver numbers + commits + git tags
- Everything that should be published is built and published appropriately

Good luck!